documentation/docs/05-misc/02-testing.md

```diff
@@ -2,7 +2,155 @@
 title: Testing
 ---
 
-- component testing basics
-- rune testing basics
-- vitest setup
-- e2e
+Testing helps you write and maintain your code and guard against regressions. Testing frameworks help you with that, allowing you to describe assertions or expectations about how your code should behave. Svelte is unopinionated about which testing framework you use, so you can write unit tests, integration tests, and end-to-end tests using solutions like Vitest, Jasmine, Cypress and Playwright.
+
+## Unit/Integration-testing using Vitest
+
+Unit tests allow you to test small isolated parts of your code. Integration tests allow you to test parts of your application to see if they work together. We recommend using [Vitest](https://vitest.dev/) for these kinds of tests (this assumes you use Vite for bundling your application).
+
+To get started, install Vitest (`npm install vitest -D`) and JSDOM (a library that shims DOM APIs, `npm install jsdom -D`). Then adjust your `vite.config.js` such that you can simulate a browser environment for testing Svelte components on the client:
+
+```js
+/// file: vitest.config.js
+import { defineConfig } from 'vitest/config';
+
+export default defineConfig({
+	plugins: [
+		/* ... */
+	],
+	test: {
+		// If you are testing components client-side, you need to setup a DOM environment.
+		// If not all your files should have this environment, you can use a
+		// `// @vitest-environment jsdom` comment at the top of the test files instead.
+		environment: 'jsdom'
+	},
+	// Tell Vitest to prefer the browser condition. Svelte uses a so-called exports map
+	// which loads different code depending on the environment - we want to load the browser environment.
+	resolve: process.env.VITEST
+		? {
+				conditions: ['browser']
+			}
+		: undefined
+});
+```
+
+After that, you can create a test file in which you import the component to test, then interact with it programmatically and write expectations about the results:
+
+```js
+/// file: component.test.js
+import { flushSync, mount, unmount } from "svelte";
+import { expect, test } from "vitest";
+import Component from "./Component.svelte";
+
+test("Component", () => {
```

Comment:
if we want to recommend/document component testing, it should be using testing-library/svelte .

Reply:
I think it's important to have a low level example to not immediately abstract away stuff. Then they have an easier time reasoning about the abstraction

Reply:
this is too low level. people would come up with their own abstractions as the testing-library link is easy to overlook. If you want the example this prominent, testing library should at least be called out before and in a colorful block so that people use that by default. This is not something you want to bikeshed

Reply:
The reading flow is better if first the underlying primitive is shown and then the abstraction on top - I added a code example showing how to rewrite it using the testing library, to make the benefits more obvious.